gbpservice/neutron/nsf/db/nsf_db_model.py

```diff
@@ -0,0 +1,193 @@
+# Copyright (c) 2016 OpenStack Foundation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from oslo_db.sqlalchemy import models
+from oslo_utils import uuidutils
+import sqlalchemy as sa
+from sqlalchemy.ext import declarative
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import orm
+
+
+TENANT_ID_MAX_LEN = 255
+DESCRIPTION_MAX_LEN = 255
+
+
+class HasTenant(object):
+    """Tenant mixin, add to subclasses that have a tenant."""
+
+    tenant_id = sa.Column(sa.String(TENANT_ID_MAX_LEN), index=True)
+
+
+class HasId(object):
+    """id mixin, add to subclasses that have an id."""
+
+    id = sa.Column(sa.String(36),
+                   primary_key=True,
+                   default=uuidutils.generate_uuid)
+
+
+class HasStatus(object):
+    """status mixin, add to subclasses that have a status."""
+
+    status = sa.Column(sa.String(16), nullable=False, index=True)
+
+
+class HasStatusDescription(HasStatus):
+    """Status with description mixin."""
+
+    status_description = sa.Column(sa.String(DESCRIPTION_MAX_LEN))
+
+
+# Do we need anything to be overridden from modelBase ??
+class ServiceManagerBase(models.ModelBase):
+    """Base class for NSF Models."""
+
+    __table_args__ = {'mysql_engine': 'InnoDB'}
+
+    def __iter__(self):
+        self._i = iter(orm.object_mapper(self).columns)
+        return self
+
+    def next(self):
+        n = next(self._i).name
+        return n, getattr(self, n)
+
+    __next__ = next
+
+    def __repr__(self):
+        """sqlalchemy based automatic __repr__ method."""
+        items = ['%s=%r' % (col.name, getattr(self, col.name))
+                 for col in self.__table__.columns]
+        return "<%s.%s[object at %x] {%s}>" % (self.__class__.__module__,
+                                               self.__class__.__name__,
+                                               id(self), ', '.join(items))
+
+    @declarative.declared_attr
+    def __tablename__(cls):
+        return cls.__name__.lower() + 's'
+
+
+BASE = declarative_base(cls=ServiceManagerBase)
+
+
+class PortInfo(BASE, HasId, HasTenant):
+    """Represents the Port Information"""
+    __tablename__ = 'port_infos'
+
+    port_policy = sa.Column(sa.String(36))  # neutron_port, gbp_policy_target
```

Comment:
NetworkInfo is optional - at present only for HA. Will rename port_policy to port_model.